File: backend/src/main/java/com/grouply/backend/direct_message/dto/DirectMessageDTO.java
package com.grouply.backend.direct_message.dto;

import com.grouply.backend.user.Dtos.UserDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.time.Instant;

@Data
@AllArgsConstructor
@Builder
public class DirectMessageDTO {


    private Long id;
    private Long roomId;

    private UserDTO sender;

    private UserDTO recipient;

    private String message;
    private Instant sentAt;

}

File: frontend/src/components/layout/message-box/model/direct.message.dto.ts
// adjust this import path if needed

import type { UserDTO } from "../../../../dtos/models_dtos/user_dto";


export interface DirectMessageDTO {
  id: number;
  roomId: number;
  sender: UserDTO;
  recipient: UserDTO;
  message: string;
  sentAt: string; // Instant -> ISO string
}
